conans/model/info.py

```diff
@@ -443,25 +456,21 @@ def package_id(self):
         """
         text = self.dumps()
         package_id = sha1(text.encode())
-        # FIXME: Should it be done before calling this function?
-        try:
-            self.options.validate()
-        except ConanException as e:
-            self.invalid = BINARY_INVALID, str(e)
         return package_id
 
-    def serialize_min(self):
-        """
-        This info will be shown in search results.
-        """
-        # self.settings is already a simple serialized list, it has been loaded from conaninfo.txt
-        conan_info_json = {"settings": dict(self.settings),
-                           "options": dict(self.options.serialize())["options"],
-                           "requires": self.requires.serialize()
-                           }
-        return conan_info_json
-
     def header_only(self):
         self.settings.clear()
         self.options.clear()
         self.requires.clear()
+
+    def validate(self):
+        # If the options are not fully defined, this is also an invalid case
+        try:
+            self.options.validate()
+        except ConanException as e:
+            self.invalid = BINARY_INVALID, str(e)
+
+        try:
+            self.settings.validate()
+        except ConanException as e:
+            self.invalid = BINARY_INVALID, str(e)
```

Comment:
This `validate` method is fine to isolate this logic, but what I'd expect is to raise an exception too instead of only giving the value to the `self.invalid`.

Reply:
Not really, it is necessary to keep it without raising, because for the ``compatibility`` cases we want to be able to continue processing. It will only raise later, if you actually try to build or install such invalid configuration, but at this point, you can't raise yet.